### DISCUSSION TITLE
Rebuild images regularly to take Alpine security updates

### DISCUSSION THREAD

**Discussion by strugee** [Mailu/Mailu#3185] [Feature requests / Ideas]
Basically, what the title says. I gathered from https://github.com/Mailu/Mailu/issues/1182#issuecomment-535971401 that this _used_ to be something the project did, and guessed that it didn't survive the migration from Travis CI to GitHub Actions instead of being intentionally dropped.

I'm about to push a PR to do this; the primary purpose of this post is to have an issue number to reference in a changelog entry.

**Comment by nextgens** [comment]
Since then we have been bitten by alpine quite a few times and we do not do it anymore.

'stable' versions of Mailu get minor updates from alpine (with the security patches); upcoming versions get based on a different alpine branch and get newer code.

**Reply by strugee** [reply]
@nextgens apologies for the late response, work got busy and then I just kind of forgot about this :(

> 'stable' versions of Mailu get minor updates from alpine (with the security patches)

I don't think I'm quite understanding, because this is what I _thought_ the patch in #3186 was doing. Or do you mean that Mailu will cut a new A.B.x release every time Alpine pushes out security updates (in the same Alpine branch)?

I thought Alpine had stability guarantees in the same distribution branch, is that not true? Or are they just not as strict about it as e.g. Debian?

**Reply by strugee** [reply]
(FWIW I _did_ search the docs and issues and such decently extensively and wasn't really able to find a clear description of the process here, but if I'm asking questions that have been answered before I'd be grateful for just a pointer to where I can find this info)

**Reply by nextgens** [reply]
Have a look to #3195

New major Mailu version -> new alpine branch
New minor Mailu version -> bugfix and maybe security patches from the same alpine branch

**Reply by strugee** [reply]
OK so for example `2024.03.1` is `2024.03.0` but with Alpine security/bugfix updates applied? And every time there's Alpine security updates a new minor version will be manually cut, like `2024.03.2` or whatever? How are folks notified that that needs to happen?

**Reply by nextgens** [reply]
Yes.

There is no automatic notification and we do not need one; this has worked pretty well so far.

**Reply by strugee** [reply]
Got it, ok. Thanks for indulging my questions, I really really appreciate it - and so does the rest of the [SeaGL](https://seagl.org/) staff.
